File: node.js/src/original/persistence/publishers/publisherProvider.ts
import { PublisherId, Publisher } from 'src/original/domain/books/entities';
import { IPublisherProvider } from 'src/original/domain/books/publishers';
import { nonEmptyString, nonEmptyUuid } from '#core/typing';
import { PublishingHouseOrm } from '../publishingHouseOrm';

export class PublisherProvider implements IPublisherProvider {
  constructor(private orm: PublishingHouseOrm) {}

  async getById(publisherId: PublisherId): Promise<Publisher> {
    const entity = await this.orm.publishers.findById(publisherId);

    if (!entity) throw Error(`Publisher with id ${publisherId} not found`);

    const { id, name } = entity;

    return {
      id: nonEmptyUuid(id),
      name: nonEmptyString(name),
    };
  }
}

File: java/src/main/java/io/eventdriven/slimdownaggregates/original/domain/books/publishers/PublisherProvider.java
package io.eventdriven.slimdownaggregates.original.domain.books.publishers;

import io.eventdriven.slimdownaggregates.original.domain.books.entities.Publisher;
import io.eventdriven.slimdownaggregates.original.domain.books.entities.PublisherId;

public interface PublisherProvider {
  Publisher getById(PublisherId publisherId);
}
